src/main/java/org/candlepin/pki/huffman/Huffman.java

```diff
@@ -137,28 +137,34 @@ public void printTrie(HuffNode hn, int tab) {
 
     private void makePathForURL(StringTokenizer st, PathNode parent, PathNode endMarker) {
         if (st.hasMoreTokens()) {
-            String childVal = st.nextToken();
-            if (childVal.isEmpty()) {
+            String nextValue = st.nextToken();
+            if (nextValue.isEmpty()) {
                 return;
             }
             boolean isNew = true;
-            for (NodePair child : parent.getChildren()) {
-                if (child.getName().equals(childVal) &&
-                    !child.getConnection().equals(endMarker)) {
-                    makePathForURL(st, child.getConnection(), endMarker);
+            for (NodePair peer : parent.getChildren()) {
+                if (peer.getName().equals(nextValue)) {
                     isNew = false;
+                    if (!peer.getConnection().equals(endMarker)) {
+                        if (!st.hasMoreTokens()) {
+                            peer.getConnection().getChildren().removeAll(peer.getConnection().getChildren());
```

Comment:
We can use `peer.getConnection().getChildren().clear()` here

Reply:
done